toolchains/sysimage/toolchain.bzl

```diff
@@ -2,6 +2,27 @@
 Tools for building IC OS image.
 """
 
+def run_with_icos_build_wrapper(command, escape_dollars = False):
+    """Wraps a command in a temporary directory and sets the ICOS_TEMP_DIR environment variable.
+
+    Args:
+        command: The command to be wrapped.
+        escape_dollars: Whether to escape dollar signs in the command. This is necessary in
+        contexts where Bazel expands dollar-expressions.
+
+    Returns:
+        The wrapper command.
+      """
+
+    wrapper = """
+        tmpdir=$(mktemp -d "/tmp/icosbuildXXXX") &&
+        trap 'sudo rm -rf "$tmpdir"' INT TERM EXIT &&
+        ICOS_TEMP_DIR="$tmpdir" %s "$@" """
```

Comment:
I wonder if we should use `$TMPDIR` directly. `mkdir`, and python and Rust's `tempfile` all respect this variable, but it does make it harder to find the wrapper. I don't know how much of a concern that is, though.

Reply:
Absolutely. I ended up passing --tmpdir to mktemp which according to the documentation `uses $TMPDIR if set, else /tmp`.